### PR TITLE
Add ability to specify custom configs at the cmd line

### PR DESCRIPTION
**Pull request by vgokhale**
Retains the ability to run the existing benchmark suite.

### CODE REVIEW

**Comment by micmelesse** [comment]
Let me try this out.

**Review by micmelesse — COMMENTED** [review]
ok if just run the command, I get this error. We should probably have the same default behavior?

``` root@banff-cyxtera-s80-2:~/triton_rocm# python ./python/perf-kernels/flash-attention.py
Traceback (most recent call last):
  File "/root/triton_rocm/./python/perf-kernels/flash-attention.py", line 1336, in <module>
    sys.exit(main())
  File "/root/triton_rocm/./python/perf-kernels/flash-attention.py", line 1333, in main
    run_benchmark(custom_config)
  File "/root/triton_rocm/./python/perf-kernels/flash-attention.py", line 1295, in run_benchmark
    bench_flash_attention.run(save_path=".", print_data=True)
  File "/root/triton_rocm/python/triton/testing.py", line 341, in run
    result_dfs.append(self._run(bench, save_path, show_plots, print_data, **kwargs))
  File "/root/triton_rocm/python/triton/testing.py", line 287, in _run
    ret = self.fn(**x_args, **{bench.line_arg: y}, **bench.args, **kwrags)
  File "/root/triton_rocm/./python/perf-kernels/flash-attention.py", line 1273, in bench_flash_attention
    q, k, v, input_metadata = input_helper(BATCH, HQ, HK, N_CTX_Q, N_CTX_K, D_HEAD, dtype)
  File "/root/triton_rocm/./python/perf-kernels/flash-attention.py", line 1021, in input_helper
    sm_scale = D_HEAD**-0.5
ZeroDivisionError: 0.0 cannot be raised to a negative power
```

**Comment by vgokhale** [comment]
> We should probably have the same default behavior or maybe just ask for configs?

Thanks for testing this Michael. I've specified a default of 128 which is what we used.

File: python/perf-kernels/flash-attention.py [hunk 1]
```diff
@@ -1153,172 +1164,174 @@ def test_op_bwd(Z, H, N_CTX, D_HEAD, dtype=torch.float16):
...
+    line_names = 'Time (ms)' if print_time else 'TFLOPS'
+    configs.append(triton.testing.Benchmark(
+        x_names=x_names,
+        x_vals=x_vals_list,
+        line_arg='provider',
+        line_vals=['triton'],
+        line_names=[line_names],
+        styles=[('red', '-')],
+        ylabel='ms',
+        plot_name=f'fused-attention-varlen-{mode}-d{head_size}',
```
Comment (micmelesse) [inline python/perf-kernels/flash-attention.py:1238]: Did you want to surround `varlen` with `{}`?

**Review by micmelesse — COMMENTED** [review]
I fixed a typo that was causing the test to fail.

**Pull request merged**